### PR TITLE
[Horizon] Ai quiz design

### PR DESCRIPTION
**Pull request by Ahmed-Naguib93**
Ai quiz design

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Horizon Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Build failed, skipping coverage check</td>
    </tr>
  </tbody>
</table>



<details><summary>:x: XCTest failed: CoreTests/CourseSyncCleanupInteractorTests/testSharedOfflineDirectoryForUserDeleted</summary>


    XCTAssertFalse failed
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 7b7181496f0ccc69ee8891b4662cc15886ba1f14
</p>

File: Horizon/Horizon/Sources/Features/AssignmentDetails/View/AssignmentDetailsViewModel.swift [hunk 1]
```diff
@@ -147,7 +136,7 @@ final class AssignmentDetailsViewModel {
         aiEvents.sink {  [weak self] event, controller in
             switch event {
             case .assist:
-                self?.appEnvironment.router.route(to: "/tutor", from: controller)
+                self?.appEnvironment.router.show(AIAssembly.makeAITutorView(), from: controller, options: .modal(isDismissable: false))
```
Comment (szabinst) [inline Horizon/Horizon/Sources/Features/AssignmentDetails/View/AssignmentDetailsViewModel.swift:139]: let's use `router.route(to:)` function and now that appEnvironment is not used anymore, we can delete the local variable and inject router directly to the vm

**Review by szabinst — CHANGES_REQUESTED** [review]
LGTM! Small crs below:

File: Horizon/Horizon/Sources/Features/AI/AIQuiz/View/AIQuizView.swift [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+            viewModel.regenerateQuiz()
+        }) {
+            Text("Regenerate", bundle: .horizon)
+                .underline()
+                .foregroundColor(.textLightest)
+                .font(.regular16)
+        }
+    }
+
+    private var tryAginButton: some View {
```
[the remark this answers is not in the document]
Comment (Ahmed-Naguib93) [reply Horizon/Horizon/Sources/Features/AI/AIQuiz/View/AIQuizView.swift]: Changed

File: Horizon/Horizon/Sources/Features/AssignmentDetails/View/AssignmentDetailsViewModel.swift (hunk 1 above)
Comment (Ahmed-Naguib93) [reply Horizon/Horizon/Sources/Features/AssignmentDetails/View/AssignmentDetailsViewModel.swift]: Done

**Review by szabinst — APPROVED** [review]

**Pull request merged**